actors/market/tests/market_actor_test.rs

```diff
@@ -495,3 +587,257 @@ fn construct_and_verify(rt: &mut MockRuntime) {
+    expect_provider_control_address(rt, provider, owner, worker);
+
+    let params = WithdrawBalanceParams { provider_or_client: provider, amount: withdraw_amount };
+
+    rt.expect_send(
+        owner,
+        METHOD_SEND,
+        RawBytes::default(),
+        expected_send.clone(),
+        RawBytes::default(),
+        ExitCode::OK,
+    );
+    let ret: WithdrawBalanceReturn = rt
+        .call::<MarketActor>(Method::WithdrawBalance as u64, &RawBytes::serialize(params).unwrap())
+        .unwrap()
+        .deserialize()
+        .unwrap();
+    rt.verify();
+
+    assert_eq!(
+        expected_send, ret.amount_withdrawn,
+        "return value indicates {} withdrawn but expected {}",
+        ret.amount_withdrawn, expected_send
+    );
+}
+
+fn withdraw_client_balance(
+    rt: &mut MockRuntime,
+    withdraw_amount: TokenAmount,
+    expected_send: TokenAmount,
+    client: Address,
+) {
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, client);
+    rt.expect_send(
+        client,
+        METHOD_SEND,
+        RawBytes::default(),
+        expected_send.clone(),
+        RawBytes::default(),
+        ExitCode::OK,
+    );
+    rt.expect_validate_caller_addr(vec![client]);
+
+    let params = WithdrawBalanceParams { provider_or_client: client, amount: withdraw_amount };
+
+    let ret: WithdrawBalanceReturn = rt
+        .call::<MarketActor>(Method::WithdrawBalance as u64, &RawBytes::serialize(params).unwrap())
+        .unwrap()
+        .deserialize()
+        .unwrap();
+    rt.verify();
+
+    assert_eq!(
+        expected_send, ret.amount_withdrawn,
+        "return value indicates {} withdrawn but expected {}",
+        ret.amount_withdrawn, expected_send
+    );
+}
+
+#[allow(clippy::too_many_arguments)]
+fn generate_and_publish_deal_for_piece(
+    rt: &mut MockRuntime,
+    client: Address,
+    provider: Address,
+    owner: Address,
+    worker: Address,
+    start_epoch: ChainEpoch,
+    end_epoch: ChainEpoch,
+    piece_cid: Cid,
+    piece_size: PaddedPieceSize,
+) -> DealID {
+    // generate deal
+    let storage_per_epoch = BigInt::from(10u8);
+    let client_collateral = TokenAmount::from(10u8);
+    let provider_collateral = TokenAmount::from(10u8);
+
+    let deal = DealProposal {
+        piece_cid,
+        piece_size,
+        verified_deal: true,
+        client,
+        provider,
+        label: "label".to_string(),
+        start_epoch,
+        end_epoch,
+        storage_price_per_epoch: storage_per_epoch,
+        provider_collateral,
+        client_collateral,
+    };
+
+    // add funds
+    add_provider_funds(rt, deal.provider_collateral.clone(), provider, owner, worker);
+    add_participant_funds(rt, client, deal.client_balance_requirement());
+
+    // publish
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, worker);
+    let deal_ids = publish_deals(rt, provider, owner, worker, &[PublishDealReq { deal }]);
+    deal_ids[0]
+}
+
+struct PublishDealReq {
+    deal: DealProposal,
```

Comment:
nit: Unless I'm missing something I think this wrapping struct is a relic of history in the go tests and you can just pass in an array of deals unwrapped.